File: 剑指offer/3. 数组中重复的数字/3. 数组中重复的数字.py
from typing import List


class Solution:
    def findRepeatNumber(self, nums: List[int]) -> int:
        result = -1
        if not nums:
            return result
        aset = set()
        for num in nums:
            if num in aset:
                result = num
            aset.add(num)
        return result


if __name__ == '__main__':
    nums = [2, 3, 1, 0, 2, 5, 3]
    print(Solution().findRepeatNumber(nums))

File: 剑指offer/3. 数组中重复的数字/3. 数组中重复的数字.go
package main

func findRepeatNumber(nums []int) int {
	result := -1
	if len(nums) == 0{
		return 0
	}
	mapping := make(map[int]int, len(nums))
	for _, num := range nums{
		if _, ok := mapping[num]; ok{
			result = num
		}
		mapping[num] = num
	}
	return result
}

func main() {
		nums := []int{2, 3, 1, 0, 2, 5, 3}
		println(findRepeatNumber(nums))
}